packages/lexical-react/src/LexicalTableOfContents.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import type {NodeKey, NodeMutation} from 'lexical';
+
+import {$getNodeByKey, $getRoot, TextNode} from 'lexical';
+import {$isHeadingNode, HeadingNode} from 'packages/lexical-rich-text/src';
+import {useEffect, useState} from 'react';
+
+import {useLexicalComposerContext} from './LexicalComposerContext';
+
+function $insertHeadingIntoTableOfContents(
+  prevHeading: HeadingNode | null,
+  newHeading: HeadingNode | null,
+  currentTableOfContents: Array<[NodeKey, string, string]>,
+): Array<[NodeKey, string, string]> {
+  if (!newHeading) {
+    return currentTableOfContents;
+  }
+  const newNode: [NodeKey, string, string] = [
```

Comment:
Let's not call this node, node means a completely different thing
```suggestion
  const newEntry: [NodeKey, string, string] = [
```